### PR TITLE
simple client wrap around grpc

### PR DESCRIPTION
**Pull request by yutongp**
@xiang90:
1. do you prefer interface?
2. are we gonna use `pb.FileInfo` as file info struct or define `cfsFileinfo`? 
3. I am not familiar with protobuf, can we define `Error()` for `pb.Error` struct?

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	pb "github.com/c-fs/cfs/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type Client struct {
+	conn *grpc.ClientConn
+}
+
+func New(address string) *Client {
```
Comment (xiang90) [inline client/client.go]: Return an error please. We normally do not fatal or panic inside a pkg when possible. 
Especially in this case, the user of pkg cannot know if the address is available/reachable before calling New(). Fatal is not a good option here.

File: client/client.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	conn, err := grpc.Dial(address)
+	if err != nil {
+		log.Fatalf("Cannot connect: %v", err)
+	}
+	return &Client{conn: conn}
+}
+
+func (c *Client) Write(
```
Comment (xiang90) [inline client/client.go]: one line please.

File: client/client.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	pbClient := pb.NewCfsClient(c.conn)
+	reply, err := pbClient.Write(
+		ctx,
+		&pb.WriteRequest{Name: name, Offset: offset, Data: data, Append: isAppend},
+	)
+
+	if err != nil {
+		return 0, err
+	}
+	return reply.BytesWritten, pbErr2GoErr(reply.Error)
```
Comment (xiang90) [inline client/client.go]: parseErr(reply.Error)

File: client/client.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+func (c *Client) Read(
```
Comment (xiang90) [inline client/client.go]: one line. there is no 80 line requirement in go. if you use far more than 80 chars, then there is something wrong with your approach. :)

File: client/client.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	"errors"
+	"log"
+
+	pb "github.com/c-fs/cfs/proto"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+type Client struct {
+	conn *grpc.ClientConn
```
Comment (xiang90) [inline client/client.go]: why not keep the grpc client? we create the client for each function call right now.

**Comment by xiang90** [comment]
> do you prefer interface?
> are we gonna use pb.FileInfo as file info struct or define cfsFileinfo?
> I am not familiar with protobuf, can we define Error() for pb.Error struct?
1. hm... i do not really care for now. we can add the interface when we need.
2. FileInfo is an interface in go actually. We can define cfs.FileInfo. But let us use pb.FileInfo for now?
3. I am not sure. I can check for you.

File: client/client.go (hunk 5 above)
Comment (yutongp) [inline client/client.go]: My fault here...

File: client/client.go (hunk 4 above)
Comment (yutongp) [inline client/client.go]: Okay. In my company, we do the wrap line. I thought, it is some how a standard.

**Comment by yutongp** [comment]
I think later on we may define cfs.FileInfo and expose it as an os.FileInfo interface.

**Comment by yutongp** [comment]
issue https://github.com/c-fs/cfs/issues/15

**Pull request merged**